### ISSUE TITLE
webpack library configuration does not resolve source map sources

### ISSUE THREAD

**Issue by dgozman** [microsoft/vscode-pwa#60]
Here is a sample webpack configuration file

```
module.exports = {
  target: 'node',
  devtool: 'source-map',
  entry: './src/lib.ts',
  module: {
    rules: [
      {
        loader: 'ts-loader',
        exclude: /node_modules/
      }
    ]
  },
  node: {
    __dirname: false,
  },
  resolve: {
    extensions: [ '.ts', '.js' ]
  },
  externals: [nodeExternals()],
  output: {
    filename: 'lib.js',
    path: path.resolve(__dirname, 'build'),
    library: 'lib',
    libraryTarget:'umd'
  }
};
```

This produces source map source urls which look like `webpack://lib/./src/lib.ts` and should be mapped to `src/lib.ts`.

Here is a [sample fix](https://github.com/dgozman/vscode-pwa/commit/d81febb82695b4d48b209bb91192f33ebf5d57ab) against an outdated branch.

**Comment by connor4312** [comment]
@dgozman it looks like your fork is private, can you make it public or push the fix to a branch on this repo so we can see it?

**Comment by connor4312** [comment]
Thanks, I can see it now!

You might have seen we moved over this handling to the user-configurable `sourceMapOverrides` config matching the existing adapters, which doesn't support full regexes right now. I'm not sure how/whether this scenario works with the current adapters, but I was able to reproduce the issue with PWA using the sample code you provided. I think my next step would be to first see if this reproduces with the existing adapters; if it doesn't, then there's something we missed in porting the logic to PWA, if it does then we should probably figure out how to take your linked fix into the `sourceMapOverrides`-driven world.

This is endgame week for vscode so I don't anticipate getting too much headway on this issue this week. If you have time to look at applying the fix to the current repo, that would be great, otherwise I'll look into this issue late this week or, more likely, next week 🙂
